### PR TITLE
feat: Add additional functionality for autopilot clusters

### PR DESCRIPTION
**Pull request by kliu47**
Additionally GA functionality for GKE autopilot modules:

- [GCS Fuse addition](https://cloud.google.com/kubernetes-engine/docs/how-to/persistent-volumes/cloud-storage-fuse-csi-driver)
- [Bakeup for GKE](https://cloud.google.com/kubernetes-engine/docs/add-on/backup-for-gke/concepts/backup-for-gke)
  - [Autopilot Support](https://cloud.google.com/kubernetes-engine/docs/add-on/backup-for-gke/concepts/about-autopilot)
- [Stateful HA Operator](https://cloud.google.com/kubernetes-engine/docs/how-to/stateful-ha)

### CODE REVIEW

**Comment by kliu47** [comment]
Updated the test in simple_autopilot_public as an example of enabling gcs_fuse

File: autogen/main/cluster.tf.tmpl [hunk 1]
```diff
@@ -324,6 +324,13 @@ resource "google_container_cluster" "primary" {
       }
     }
 
+
```
Comment (apeabody) [inline autogen/main/cluster.tf.tmpl:327]: Let's minimize the white space

```suggestion
```

File: autogen/main/cluster.tf.tmpl [hunk 2]
```diff
@@ -324,6 +324,13 @@ resource "google_container_cluster" "primary" {
...
+
```
Comment (apeabody) [inline autogen/main/cluster.tf.tmpl:328]:
```suggestion
```

File: autogen/main/cluster.tf.tmpl [hunk 3]
```diff
@@ -347,11 +354,6 @@ resource "google_container_cluster" "primary" {
         enabled = stateful_ha_config.value.enabled
       }
     }
-
-    config_connector_config {
-      enabled = var.config_connector
-    }
-    {% endif %}
     {% if beta_cluster and autopilot_cluster != true %}
```
Comment (apeabody) [inline autogen/main/cluster.tf.tmpl:357]: Let's add one whitespace back

```suggestion

    {% if beta_cluster and autopilot_cluster != true %}
```

File: autogen/main/variables.tf.tmpl [hunk 4]
```diff
@@ -759,7 +760,7 @@ variable "stateful_ha" {
   default     = false
 }
 
-{% endif %}
+
```
Comment (apeabody) [inline autogen/main/variables.tf.tmpl:763]:
```suggestion
```

**Review by apeabody — CHANGES_REQUESTED** [review]
Thanks for the contribution @kliu47!

I've noted some initial feedback on whitespace and triggered the CI tests.

/gcbrun

File: autogen/main/cluster.tf.tmpl (hunk 3 above)
Comment (kliu47) [reply autogen/main/cluster.tf.tmpl:356]: Fixed

File: autogen/main/variables.tf.tmpl (hunk 4 above)
Comment (kliu47) [reply autogen/main/variables.tf.tmpl]: Fixed

File: autogen/main/cluster.tf.tmpl (hunk 2 above)
Comment (kliu47) [reply autogen/main/cluster.tf.tmpl]: Fixed

**Comment by apeabody** [comment]
From the linter:

```
Checking submodule's files generation
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=.git' /workspace/modules/instance_template/README.md /tmp/tmp.WhX65XKh44/workspace/modules/instance_template/README.md
23c23
< | description | The template's description | `""` |`string` | no |
---
> | description | The template's description | `string` | `""` | no |
33c33
< | instance\_description | Description of the generated instances | `""` | `string` | no |
---
> | instance\_description | Description of the generated instances | `string` | `""` | no |
Error: submodule's files generation has not been run, please run the
'make build' command and commit changes
/usr/local/bin/task_helper_functions.sh: line 30: DELETE_AT_EXIT: readonly variable
Checking for documentation generation
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=autogen' '--exclude=*.tfvars' '--exclude=*metadata.yaml' /workspace/modules/instance_template/README.md /tmp/tmp.IrquMowoNu/generate_docs/workspace/modules/instance_template/README.md
23c23
< | description | The template's description | `""` |`string` | no |
---
> | description | The template's description | `string` | `""` | no |
33c33
< | instance\_description | Description of the generated instances | `""` | `string` | no |
---
> | instance\_description | Description of the generated instances | `string` | `""` | no |
Error: Documentation generation has not been run, please run the
'make docker_generate_docs' command and commit the above changes.
```

**Comment by kliu47** [comment]
Weird that the linter is asking me to re-run make build and make generate docs. Tried both and VCS is showing no changes to commit.

**Comment by apeabody** [comment]
> Weird that the linter is asking me to re-run make build and make generate docs. Tried both and VCS is showing no changes to commit.

Interesting, I don't think your PR should impact those lines.  Re-running

/gcbrun

**Review by apeabody — APPROVED** [review]
Thanks for the contribution @kliu47!

**Pull request merged**